pkg/bootstrap/config.go

```diff
@@ -141,11 +145,11 @@ func (cfg Config) toTemplateParams() (map[string]any, error) {
 			option.DNSLookupFamily(option.DNSLookupFamilyIPv6))
 	} else {
 		// Dual Stack
-		if features.EnableDualStack {
+		if DualStackEnv {
 			// If dual-stack, it may be [IPv4, IPv6] or [IPv6, IPv4]
 			// So let the first ip family policy to decide its DNSLookupFamilyIP policy
-			netIP, _ := netip.ParseAddr(cfg.Metadata.InstanceIPs[0])
-			if netIP.Is6() && !netIP.IsLinkLocalUnicast() {
+			ipFamily := network.CheckIPFamilyTypeForFirstIPs(cfg.Metadata.InstanceIPs)
+			if ipFamily == network.IPv6 {
```

Comment:
For ipv4, you set ipv4 prefered in cluster builder, but `All` here

Reply:
Hi @hzxuzhonghu,  thanks, this should be corrected.

Reply:
Its still ALL here but AUTO in cluster_builder.

Reply:
Hi @howardjohn , I change both cluster builder and here are same as using `Cluster_ALL`

Reply:
Hi @howardjohn , I change both cluster builder and here are same as using `Cluster_ALL`, thanks for the comments.

Reply:
Hi @howardjohn, **this must match cluster_builder.go** do you mean that they should be the same in both cluster builder.go and bootstrap? And not just `DNSLookupFamily` and includes others.
I think the `DNSLookupFamily` is the same to use `Cluster_ALL`, but there should be `Localhost` and `Wildcard` in `bootstrap`. Do you mean that these should be the same as well?

Reply:
the dns lookup family is *not* the same in this PR. it should be.

Reply:
Hi @howardjohn,  I can explain about the order of IP addresses here. The major different are below:
In `cluster builder.go`, if Dual Stack is enabled, there are 2 respectively IP addresses for `Localhost` and `Wildcard`.
In `bootstrap`, there always is only 1 respectively IP addresses for `Localhost` and `Wildcard`.

Reply:
I am talking only about dns lookup family, nothing to do with localhost or wildcard.

Reply:
Hi @howardjohn,  I can explain about the order of IP addresses here. The major different are below:
In `cluster builder.go`, if Dual Stack is enabled, there are 2 respectively IP addresses for `Localhost` and `Wildcard`.
In `bootstrap`, there always is only 1 respectively IP addresses for `Localhost` and `Wildcard`.
That's why I have to find the first IP address's family to set in `bootstrap`.  Hope this explanation is clear. 
And I believe that the bootstrap configuration for Envoy should be changed to make them entirely identical once the Dual Stack feature can be alpha, beta or stable. Does it make sense?

Reply:
Hi @howardjohn,  I can explain more about the order of IP addresses here. The major different are below:

- In `cluster builder.go`, if Dual Stack is enabled, there are 2 respectively IP addresses for `Localhost` and `Wildcard`.
- In `bootstrap`, there always is only 1 respectively IP addresses for `Localhost` and `Wildcard`.

That's why I have to find the first IP address's family to set in `bootstrap`.  Hope this explanation is clear. 
And I believe that the bootstrap configuration for Envoy should be changed to make them entirely identical once the Dual Stack feature can be alpha, beta or stable. Does it make sense?